app/packs/src/components/InventoryTab.js

```diff
@@ -0,0 +1,926 @@
+/* eslint-disable object-shorthand, no-trailing-spaces, 
+object-property-newline, semi, react/no-unused-prop-types, react/prop-types */
+import React from 'react';
+import PropTypes from 'prop-types';
+import { FormGroup, ControlLabel, FormControl, Button, OverlayTrigger, Tooltip, Well, ButtonToolbar, ListGroup, ListGroupItem } from 'react-bootstrap';
+import Select from 'react-select';
+import SVG from 'react-inlinesvg';
+import InventoryFetcher from './fetchers/InventoryFetcher';
+
+export default class InventoryTab extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      inventory: undefined,
+      safetySheets: [],
+      checkSaveIconThermofischer: true,
+      checkSaveIconMerck: true,
+      vendorValue: 'All',
+      vendorSafetyPhrasesValue: '',
+      vendorChemPropertiesValue: '',
+      queryOption: 'Common Name',
+      languageOfSdd: 'en',
+      safetyPhrases: ''
+    };
+    this.handleFieldChanged = this.handleFieldChanged.bind(this);
+  }
+
+  componentDidMount() {
+    const { element } = this.props;
+    this.fetchInventory(element);
+  }
+
+  fetchInventory(element) {
+    if (element === undefined || element.is_new) {
+      return;
+    }
+    InventoryFetcher.fetchByInventoriableId(element.id, element.type).then((inventory) => {
+      if (inventory !== null) {
+        this.setState({ inventory: inventory });
+      } 
+    }).catch((errorMessage) => {
+      console.log(errorMessage);
+    });
+  }
+
+  handleFieldChanged(parameter, value) {
+    const { inventory } = this.state;
+    if (inventory) {
+      inventory.invenParameters(parameter, value);
+    }
+    this.setState({ inventory });
+  }
+
+  handleSubmitSave() {
+    const { inventory } = this.state;
+    const { element } = this.props;
+    if (!element || !inventory) {
+      return;
+    }
+    const inventoryParameters = inventory._inventory_parameters;
+    const params = { 
+      inventory_parameters: inventoryParameters,
+      inventoriable_id: element.id, 
+      inventoriable_type: element.type 
+    };
+    if (inventory.isNew) {
+      InventoryFetcher.create(params).then((newinventory) => {
+        if (newinventory) {
+          this.setState({ inventory });
+        }
+      }).catch((errorMessage) => {
+        console.log(errorMessage);
+      });
+      inventory.isNew = false;
```

Comment:
Question: Why is this line necessary? From what I can see, inventory is a local variable fetched from the component's state. In line 67 you overwrite the state with the return value of the fetcher, so the old inventory object referenced by this variable is obsolete and no longer used in the context of either the state or this method. So therefore I think this line is redundant.